### PR TITLE
Added support for configuring additional properties to be printed in the prisma schema datasource

### PR DESCRIPTION
**Pull request by willemmulder**
This allows us for example to define `referentialIntegrity = "prisma"` which is required for databases like PlanetScale

### CODE REVIEW

**Comment by willemmulder** [comment]
Is this something I can fix? I don't see any specific error, and also on my local machine I get a generic "error Command failed with exit code 1"...

**Comment by borisno2** [comment]
> Is this linting error something I can fix? I don't see any specific error, and also on my local machine I get a generic "error Command failed with exit code 1"...

Hi @willemmulder... Thanks for the PR!! Running `yarn lint` on your local machine will show you the files that need fixing, and running `yarn format` will format those files for you.

File: packages/core/src/lib/core/prisma-schema.ts [hunk 1]
```diff
@@ -191,19 +191,24 @@ function assertDbFieldIsValidForIdField(
...
   lists: Record<string, InitialisedList>,
   provider: DatabaseProvider,
-  prismaPreviewFeatures: readonly string[] | undefined
+  prismaPreviewFeatures?: readonly string[] | null,
+  additionalPrismaDatasourceProperties?: { [key: string]: string; } | null
 ) {
+  let additionalDataSourceString = '';
+  if (additionalPrismaDatasourceProperties && Object.keys(additionalPrismaDatasourceProperties).length) {
+    additionalDataSourceString = Object.keys(additionalPrismaDatasourceProperties).reduce((previousValue, key) => previousValue + `\n    ${key} = "${additionalPrismaDatasourceProperties[key]}"`,'');
+  }
```
Comment (dcousens) [inline packages/core/src/lib/core/prisma-schema.ts:200]:
```suggestion
  const additionalDataSourceString = Object.entries(additionalPrismaDatasourceProperties || {}).map(([key, value]) => `\n    ${key} = "${value}"`).join('');

```

**Comment by willemmulder** [comment]
> > Is this linting error something I can fix? I don't see any specific error, and also on my local machine I get a generic "error Command failed with exit code 1"...
> 
> Hi @willemmulder... Thanks for the PR!! Running `yarn lint` on your local machine will show you the files that need fixing, and running `yarn format` will format those files for you.

Thanks! The `yarn lint` command does not provide any useful error message on my PC (still the "error Command failed with exit code 1") but `yarn format` does the trick! :-)

**Comment by dcousens** [comment]
@willemmulder that [shouldn't happen next time](https://github.com/keystonejs/keystone/pull/7784)

**Comment by willemmulder** [comment]
> @willemmulder that [shouldn't happen next time](https://github.com/keystonejs/keystone/pull/7784)

Ah nice. Thanks!

**Review by dcousens — APPROVED** [review]

**Comment by willemmulder** [comment]
Thanks for merging! I'll be trying out PlanetScale first thing once the new release is out 👍

**Pull request merged**